packages/omgidl-parser/src/IDLNodes/UnionIDLNode.ts

```diff
@@ -1,13 +1,17 @@
 import { IDLNode } from "./IDLNode";
 import { StructMemberIDLNode } from "./StructMemberIDLNode";
-import { AnyIDLNode, IUnionIDLNode } from "./interfaces";
+import { AnyIDLNode, IEnumIDLNode, ITypedefIDLNode, IUnionIDLNode } from "./interfaces";
 import { UnionASTNode } from "../astTypes";
 import { INTEGER_TYPES, SIMPLE_TYPES } from "../primitiveTypes";
 import { Case, IDLMessageDefinition, IDLMessageDefinitionField } from "../types";
 
 export class UnionIDLNode extends IDLNode<UnionASTNode> implements IUnionIDLNode {
+  private switchTypeNeedsResolution = false;
   constructor(scopePath: string[], astNode: UnionASTNode, idlMap: Map<string, AnyIDLNode>) {
     super(scopePath, astNode, idlMap);
+    if (!SIMPLE_TYPES.has(this.astNode.switchType)) {
```

Comment:
Good question! It's actually disallowed by the grammar since these are considered keywords by the parser. So it will error saying that it expected a `%NAME` instead of `uint8`.

Reply:
thank goodness